client/my-sites/backup/backup-contents-page/file-browser/util.ts

```diff
@@ -0,0 +1,28 @@
+import { BackupLsResponse, FileBrowserItem } from './types';
+
+export const parseBackupContentsData = ( payload: BackupLsResponse ): FileBrowserItem[] => {
+	if ( ! payload || ! payload.contents || ! payload.ok ) {
+		return [];
+	}
+
+	const transformedData = Object.entries( payload.contents ).map( ( [ name, item ] ) => {
+		return {
```

Comment:
We may be able to do this here:

```
		let type = item.type;
		if ( item.type === 'file' && item.has_children ) {
			type = 'dir';
		}

		return {
			name,
			type,
			hasChildren: item.has_children ?? false,
			...( item.period && { period: item.period } ),
		};
```

Or if we're concerned about changing type, maybe we create a transform `icon` field and pass type through directly.

Reply:
Thanks for the suggestion! I moved that logic to `transformFileType()` and now we handle the file type detection based on the `fileType` and also the `extension`.

Also, for icons, given that we now have a more specific `fileType`, we just declared a dictionary with the `fileType` and the icon that needs to be rendered.